tools/RepositoryManagement/DevHome.RepositoryManagement/ViewModels/RepositoryManagementItemViewModel.cs

```diff
@@ -2,61 +2,451 @@
+        processStartInfo.ArgumentList.Add("configure");
+        processStartInfo.Verb = "RunAs";
+
+        StartProcess(processStartInfo, nameof(RunConfigurationFile));
+    }
+
+    internal RepositoryManagementItemViewModel(
+        Window window,
+        RepositoryManagementDataAccessService dataAccess,
+        IStringResource stringResource,
+        ConfigurationFileBuilder configurationFileBuilder,
+        string repositoryName,
+        string cloneLocation)
     {
-        throw new NotImplementedException();
+        _window = window;
+        _dataAccess = dataAccess;
+        _stringResource = stringResource;
+        _configurationFileBuilder = configurationFileBuilder;
+        RepositoryName = repositoryName;
+        _clonePath = cloneLocation;
     }
 
-    [RelayCommand]
     public void RemoveThisRepositoryFromTheList()
     {
-        throw new NotImplementedException();
+        var repository = GetRepositoryReportIfNull(nameof(RemoveThisRepositoryFromTheList));
+        if (repository == null)
+        {
+            return;
+        }
+
+        _dataAccess.SetIsHidden(repository, true);
+    }
+
+    private void OpenRepositoryInFileExplorer(string repositoryName, string cloneLocation, string action)
+    {
+        _log.Information($"Showing {repositoryName} in File Explorer at location {cloneLocation}");
+        TelemetryFactory.Get<ITelemetry>().Log(
+            EventName,
+            LogLevel.Critical,
+            new RepositoryLineItemEvent(action, repositoryName));
+
+        var processStartInfo = new ProcessStartInfo
+        {
+            UseShellExecute = true,
+
+            // Not catching PathTooLongException.  If the file was in a location that had a too long path,
+            // the repo, when cloning, would run into a PathTooLongException and repo would not be cloned.
+            FileName = Path.GetFullPath(cloneLocation),
+        };
+
+        StartProcess(processStartInfo, action);
+    }
+
+    private void OpenRepositoryinCMD(string repositoryName, string cloneLocation, string action)
+    {
+        _log.Information($"Showing {repositoryName} in CMD at location {cloneLocation}");
+        TelemetryFactory.Get<ITelemetry>().Log(
+            EventName,
+            LogLevel.Critical,
+            new RepositoryLineItemEvent(action, repositoryName));
+
+        var processStartInfo = new ProcessStartInfo
+        {
+            UseShellExecute = true,
+
+            // Not catching PathTooLongException.  If the file was in a location that had a too long path,
+            // the repo, when cloning, would run into a PathTooLongException and repo would not be cloned.
+            FileName = "CMD",
+            WorkingDirectory = Path.GetFullPath(cloneLocation),
+        };
+
+        StartProcess(processStartInfo, action);
+    }
+
+    private void StartProcess(ProcessStartInfo processStartInfo, string operation)
+    {
+        try
+        {
+            Process.Start(processStartInfo);
```

Comment:
should we wait for the process to exit and then read the stdOutput/StdErr, to check for errors?

Reply:
Added as a TODO.  I don't know what text DevHome should look for.  I can talk to Amir about that.